dotnet/src/SemanticKernel.Test/Diagnostics/VerifyTest.cs

```diff
@@ -0,0 +1,144 @@
+    [Fact]
+    public void GreaterThanPassesWhenValueIsGreaterThanTarget()
+    {
+        Verify.GreaterThan<int>(1, 0);
+        Verify.GreaterThan<double>(1.0, 0.0);
+        Verify.GreaterThan<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue);
+    }
+    #endregion
+
+    #region Verify.GreaterThanOrEqualTo
+    [Fact]
+    public void GreaterThanOrEqualToThrowsWhenValueIsLessThanTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.GreaterThanOrEqualTo<int>(-1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<double>(-0.1, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.GreaterThan<TimeSpan>(TimeSpan.MinValue, TimeSpan.MaxValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    [Fact]
+    public void GreaterThanOrEqualToPassesWhenValueIsGreaterThanOrEqualToTarget()
+    {
+        Verify.GreaterThanOrEqualTo<int>(1, 0);
+        Verify.GreaterThanOrEqualTo<double>(0.0, 0.0);
+        Verify.GreaterThanOrEqualTo<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue);
+    }
+    #endregion
+
+    #region Verify.LessThan
+    [Fact]
+    public void LessThanThrowsWhenValueIsGreaterThanOrEqualToTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.LessThan<int>(1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThan<double>(0.0, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThan<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    public void LessThanPassesWhenValueIsLessThanTarget()
+    {
+        Verify.LessThan<int>(-1, 0);
+        Verify.LessThan<double>(-0.1, 0.0);
+        Verify.LessThan<TimeSpan>(TimeSpan.MinValue, TimeSpan.MaxValue);
+    }
+    #endregion
+
+    #region Verify.LessThanOrEqualTo
+    [Fact]
+    public void LessThanOrEqualToThrowsWhenValueIsGreaterThanTarget()
+    {
+        var ex = Assert.Throws<ValidationException>(() => Verify.LessThanOrEqualTo<int>(1, 0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThanOrEqualTo<double>(0.1, 0.0));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+
+        ex = Assert.Throws<ValidationException>(() => Verify.LessThanOrEqualTo<TimeSpan>(TimeSpan.MaxValue, TimeSpan.MinValue));
+        Assert.Equal(ex.ErrorCode, ValidationException.ErrorCodes.OutOfRange);
+    }
+
+    public void LessThanOrEqualToPassesWhenValueIsLessThanOrEqualToTarget()
```

Comment:
Same, is this needed?

Reply:
This is needed since throwing exceptions on negative cases doesn't guarantee the opposite will pass if the implementation is buggy. A simple case would be the function throws exception regardless of the input.

Reply:
Shoot sorry, clearly was too sleepy. I didn't realize these were tests for the new Verify stuff >.<. Ignore me!